diff --git salt/states/mount.py salt/states/mount.py
index 9e88e2f..766f88e 100644
--- salt/states/mount.py
+++ salt/states/mount.py
@@ -198,7 +198,8 @@ def mounted(name,
                             converted_size = int(size_match.group('size_value')) * 1024 * 1024
                         opt = "size={0}k".format(converted_size)
 
-                    if opt not in active[real_name]['opts'] and opt not in active[real_name]['superopts'] and opt not in mount_invisible_options:
+                    if opt not in active[real_name]['opts'] and opt not in mount_invisible_options \
+                       and ('superopts' in active[real_name] and opt not in active[real_name]['superopts']):
                         if __opts__['test']:
                             ret['result'] = None
                             ret['comment'] = "Remount would be forced because options ({0}) changed".format(opt)
